libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/MockNetworkTests.m

```diff
@@ -29,6 +29,52 @@
 #import "CSFNetwork+Internal.h"
 #import "CSFSalesforceAction.h"
 
+@interface MockAction : CSFAction
+
+@property (nonatomic, strong) NSString *actionName;
+@property (nonatomic, strong) dispatch_semaphore_t semaphore;
+
+- (instancetype) initWithName:(NSString*)actionName;
+
+@end
+
+@implementation MockAction
```

Comment:
Mock action that does an actual http call (to /services/data so it doesn't need an access token).
However the completeOperationWithError is overridden, it waits on a semaphore before calling its super.
The semaphore can be released by calling completeAction. That way we can stop the action in flight.